modules/javafx.base/src/main/java/com/sun/javafx/collections/VetoableListDecorator.java

```diff
@@ -100,6 +101,20 @@ public void removeListener(InvalidationListener listener) {
         helper = ListListenerHelper.removeListener(helper, listener);
     }
 
+    @Override
+    public final void doSort(Comparator<? super E> comparator) {
+        var sortedList = new ArrayList<>(list);
+        sortedList.sort(comparator);
+        onProposedChange(Collections.unmodifiableList(sortedList), 0, size());
+        try {
+            modCount++;
+            list.setAll(sortedList);
+        } catch(Exception e) {
+            modCount--;
+            throw e;
+        }
```

Comment:
Could this be replaced by `setAll(sortedList)`?

Reply:
That would incur another unnecessary copy of the list. I've factored out the common `setAllImpl` instead to prevent these copies.